contracts/modules/interfaces/IexecPoco2.v8.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+interface IexecPoco2 {
+    event AccurateContribution(address indexed worker, bytes32 indexed taskId);
+    event FaultyContribution(address indexed worker, bytes32 indexed taskId);
+
+    event TaskInitialize(bytes32 indexed taskId, address indexed workerpool);
+    event TaskContribute(bytes32 indexed taskId, address indexed worker, bytes32 hash);
+    event TaskConsensus(bytes32 indexed taskId, bytes32 consensus);
+    event TaskReveal(bytes32 indexed taskId, address indexed worker, bytes32 digest);
+    event TaskFinalize(bytes32 indexed taskId, bytes results);
+    event TaskClaimed(bytes32 indexed taskId);
+    event TaskReopen(bytes32 indexed taskId);
+
+    function initialize(bytes32 dealId, uint256 index) external returns (bytes32);
+
+    function claim(bytes32 taskId) external;
+
+    function contribute(
+        bytes32 taskId,
+        bytes32 resultHash,
+        bytes32 resultSeal,
+        address enclaveChallenge,
+        bytes memory enclaveSign,
```

Comment:
calldata

Reply:
[Use callback instead of memory](https://github.com/iExecBlockchainComputing/PoCo/pull/63/commits/227b5cfefcfb4ed3170fbd149cb074dc393a3b47)
*calldata